great_expectations/core/datasource_dict.py

```diff
@@ -173,64 +194,18 @@ def _init_block_style_datasource(
             datasource_name=name, datasource_config=config
         )
 
-
-class CacheableDatasourceDict(DatasourceDict):
```

Comment:
Didn't feel a need to have this subclass if the base class was going to have some caching behavior - instead, we can inject whatever cache we require during init (TTL or no eviction policy for Cloud vs others).